### PR TITLE
refactor: bump CocoaHTTPServer to ec6f189c15

### PR DESCRIPTION
**Pull request by KazuCocoa**

### CODE REVIEW

**Comment by KazuCocoa** [comment]
cc @qmfrederik

**Comment by mykola-mokhnach** [comment]
@qmfrederik would you like to push more changes before the component is published?

**Comment by qmfrederik** [comment]
Well, I came to the conclusion that is very tedious to make changes to CocoaHTTPServer or RoutingHTTPServer and then have them flow into WebDriverAgent.

Since both CocoaHTTPServer and RoutingHTTPServer are unmaintained projects which have been branched by Appium, the code of these projects is now kind of owned by Appium.

So I wondered whether it wouldn't be easier to just move the code of CocoaHTTPServer and RoutingHTTPServer directly into WebDriverAgent.

I've done an experiment here: https://github.com/qmfrederik/WebDriverAgent/tree/exploration/import-http-server . I tried to keep the commits small (so there are many), but it comes down to:
- Import CocoaAsyncSockets
- Import CocoaHTTPServer
- Import RoutingHTTPServer
- Remove cruft, mainly in CocoaHTTPServer (file responses, authentication, SSL support)
- Update to the latest version (7.4) of CocoaAsyncSockets 

Items for consideration are:
- It would make sense to actually take a dependency on CocoaAsyncSockets instead of copying the code, this project is actively maintained
- WebDriverAgent builds with all warnings enabled; some of the imported code generates warnings (and these have been suppressed in the source file)

A bit to my surprise, I found that the memory leak which I observed in WebDriverAgent is gone on the branch. I had to run the tests a couple of times to convince myself this is really the case :). I believe this is because these projects are so old (they target iOS 5 or so) that they are somehow misconfigured (ARC, memory management, whatever,...) when linked into a project which targets recent iOS versions. But I could be wrong.

Long story short: do you think it makes sense to consolidate the bits and pieces of CocoaHTTPServer and RoutingHTTPServer which are used by WebDriverAgent into the WebDriverAgent repository? If so, I could try to create a PR out of https://github.com/qmfrederik/WebDriverAgent/tree/exploration/import-http-server.

**Comment by mykola-mokhnach** [comment]
> A bit to my surprise, I found that the memory leak which I observed in WebDriverAgent is gone on the branch.

🎉 

> Long story short: do you think it makes sense to consolidate the bits and pieces of CocoaHTTPServer and RoutingHTTPServer which are used by WebDriverAgent into the WebDriverAgent repository?

I vote with my two hands for that. Simply create a PR and we'll merge the stuff after it passes the tests. Less dependencies is always great. @KazuCocoa would you like to add something on that?

**Comment by KazuCocoa** [comment]
Great work 🎉 

I'm also for the proposal to reduce unmaintained projects/code, and keep libraries small only we need.
Personally, it would be great to address their history like https://github.com/appium/CocoaHTTPServer#project-history a bit in our readme for respect.

**Comment by qmfrederik** [comment]
Cool! See appium/WebDriverAgent#348, CI is green :)
